webapp/src/components/common/datePicker/index.tsx

```diff
@@ -0,0 +1,98 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useRef, useState} from 'react';
+
+import './style.scss';
+import {
+    autoUpdate,
+    flip,
+    FloatingFocusManager,
+    offset,
+    shift,
+    useDismiss,
+    useFloating,
+    useInteractions,
+} from '@floating-ui/react';
+
+import {DayPicker} from 'react-day-picker';
+
+export type Props = {
+    value?: Date
+    children: React.ReactNode
+    onSelect: (date: Date) => void
+    closeOnSelect?: boolean
+};
+
+const DatePicker = ({value, children, onSelect, closeOnSelect}: Props) => {
+    const [isPopupOpen, setPopupOpen] = useState<boolean>(false);
+
+    const onSelectHandler = useCallback((day: Date) => {
+        onSelect(day);
+
+        // Checking it this way to handle null and undefined.
+        // The default behavior is to close on select.
+        if (closeOnSelect !== false) {
+            setPopupOpen(false);
+        }
+    }, [closeOnSelect, onSelect]);
+
+    const {context} = useFloating({
+        open: isPopupOpen,
+        onOpenChange: (open) => setPopupOpen(open),
+        placement: 'bottom-end',
+        whileElementsMounted: autoUpdate,
+        middleware: [
+            offset(5),
+            flip({fallbackPlacements: ['bottom-end', 'top-start', 'top-end', 'right-start', 'left-start'], padding: 5}),
+            shift(),
+        ],
+    });
+
+    const {getReferenceProps, getFloatingProps} = useInteractions([
+        useDismiss(context, {
+            enabled: true,
+            outsidePress: true,
+        }),
+    ]);
+
+    const popperRef = useRef<HTMLDivElement>(null);
+
+    return (
+        <div className='DatePicker'>
```

Comment:
Have you double checked to make sure that this doesn't conflict with any classes defined by the web app? I can't find any obvious conflict at least, but I wanted to mention it

Reply:
yes, its not used in MM webapp.